### PR TITLE
fix(TDI-46373):Resuming logs refactor with NIO and FileLock

### PR DESCRIPTION
**Pull request by pyzhou-talend**
https://jira.talendforge.org/browse/TDI-46373

### CODE REVIEW

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java [hunk 1]
```diff
@@ -508,18 +521,40 @@ public void write(String content) throws IOException {
...
+            buf.put(content.getBytes(StandardCharsets.UTF_8));
 
-            writer.write(TextQualifier);
+            buf.put(TextQualifier.getBytes(StandardCharsets.UTF_8));
 
             firstColumn = false;
         }
 
+        public FileLock getlock(){
+            try {
```
Comment (wwang-talend) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java:532]: the usage is strange for me, why works with java syn block together as a object lock, but not this mode:
```
FileLock lock;
try {
  lock = channel.lock();//or tryLock but different process
  //do things
} finally {
  lock.release();
}
```

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java [hunk 2]
```diff
@@ -495,10 +508,10 @@ public void write(String content) throws IOException {
             }
 
             if (!firstColumn) {
-                writer.write(Delimiter);
+                buf.put(Delimiter.getBytes(StandardCharsets.UTF_8));
```
[the remark this answers is not in the document]
Comment (pyzhou-talend) [reply main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java:511]: OK. I will change back to default

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java [hunk 3]
```diff
@@ -481,8 +492,10 @@ public JobLogItem(String eventDate, String type, String partName, String parentP
 
         private String lineSeparator = System.getProperty("line.separator");
 
-        public SimpleCsvWriter(Writer writer) {
-            this.writer = writer;
+        public SimpleCsvWriter(FileChannel channel) {
+            this.channel = channel;
+            buf = ByteBuffer.allocate(2<<14);//32k buffer size
+            buf.clear();
```
Comment (wwang-talend) [inline main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java:498]: seems no need the clear()

File: main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java (hunk 1 above)
Comment (pyzhou-talend) [reply main/plugins/org.talend.librariesmanager/resources/java/routines/system/ResumeUtil.java]: OK. I will do some refactoring

**Review by wwang-talend — APPROVED** [review]

**Pull request merged**